### PR TITLE
Release 9.0.0

### PR DESCRIPTION
**Pull request by mradkov**
# Changelog
### ⚠ BREAKING CHANGES

* drop following AENS delegation signature methods over the new common `createAensDelegationSignature`  implementation which accepts an object as param
   * ` delegateNamePreclaimSignature`
   * ` delegateNameClaimSignature`
   * `delegateNameTransferSignature`
   * `delegateNameRevokeSignature` 
* drop following oracle delegation signature methods over the new common  `createOracleDelegationSignature` implementation accepts an object param
   * `delegateOracleRegisterSignature`
   * `delegateOracleExtendSignature`
   * `delegateOracleRespondSignature`
* drop `waitMined` static method
* drop `assertedType`, use `decode` instead
* **validator:** recursive validator, simplify schema
* simplify buildTxHash helper
* enable verification in deep props instead of extra variable

### Features

* `payForTransaction` method ([fbf204d](https://github.com/aeternity/aepp-sdk-js/commit/fbf204dd2ba333551327adce579b9f9d9edc601b))

### Maintenance
* change default gas limit to 25000 ([831e4dd](https://github.com/aeternity/aepp-sdk-js/commit/831e4dde12a78583d1a567fdc5a9ef86518d5fe6))

### Bug Fixes

* **AENS:** name length minimum bid fee ([db6ca4c](https://github.com/aeternity/aepp-sdk-js/commit/db6ca4c3d89eedf7fa326bd6d0484ea0d94de19e))
* **delegate-signature:** stop using the default account in the contex for signing ([29d760e](https://github.com/aeternity/aepp-sdk-js/commit/29d760ef04591cd9a745a508ff4dfbb8859c347a))
* **deposit-trap:** enforce zero value for `deposit` during contract deploy ([cfb5f22](https://github.com/aeternity/aepp-sdk-js/commit/cfb5f2263316f0f70290cb174c1d334cf49c6db3))
* **mustAccountStamp:** process accounts only if supplied ([10ec2c3](https://github.com/aeternity/aepp-sdk-js/commit/10ec2c37f6f35516023cace435bdde6294786ed7))
* **payingFor:** don't check is GA if innerTx ([0ee9db4](https://github.com/aeternity/aepp-sdk-js/commit/0ee9db4b7e1cbab5b874bee56b5bf8814610b913))
* **payingFor:** don't sent to blockchain if innerTx ([523e9bb](https://github.com/aeternity/aepp-sdk-js/commit/523e9bbd62a506efd8a9d50317bcf3980052948e))
* swagger https issue ([3a876bb](https://github.com/aeternity/aepp-sdk-js/commit/3a876bbb80300cbf91891daa642f6daee61c1427))
* **tx serialisation:** accept unpackTx output produced by deserialisator ([ff0b3f5](https://github.com/aeternity/aepp-sdk-js/commit/ff0b3f551f7cf0684ac9c8e528d94eae9c31f226))
* **validator:** recursive validator, simplify schema ([95a2a23](https://github.com/aeternity/aepp-sdk-js/commit/95a2a232f3fd8b875b223e352445b045a7217dad))
* drop `assertedType`, use `decode` instead ([00d563f](https://github.com/aeternity/aepp-sdk-js/commit/00d563feca871e2c9474df667696946d1c8d12c7))
* drop `waitMined` static method ([2f299de](https://github.com/aeternity/aepp-sdk-js/commit/2f299de2c985ce2619db10e4a264f7e6d1b7279c))
* enable verification in deep props instead of extra variable ([989b36f](https://github.com/aeternity/aepp-sdk-js/commit/989b36f28e9649fd74b384f2a16bf8fd64bac85c))
* merge aens signature delegation methods into a single implementation ([456fe00](https://github.com/aeternity/aepp-sdk-js/commit/456fe00cb8b8eba09ea53800e8713550d0f329cc))
* merge oracle delegation signature methods ([88b7bf3](https://github.com/aeternity/aepp-sdk-js/commit/88b7bf3aa55b3740c9a64fb4049abcc0fdd3b277))
* simplify buildTxHash helper ([9e1fde7](https://github.com/aeternity/aepp-sdk-js/commit/9e1fde7693986359a9ac0ff80829dbc058882f7b))

### CODE REVIEW

**Comment by marc0olo** [comment]
in what ways do the breaking changes affect the devs? can we add a migration guide (like for 7.0.0 in the past) that explains what changed with examples?

**Comment by mradkov** [comment]
If https://github.com/aeternity/aesophia_http/pull/74 gets merged and deployed prior to releasing this v9.0.0 we can omit the workaround introduced in #1250.

**Comment by mradkov** [comment]
I propose to try finalizing https://github.com/aeternity/aepp-sdk-js/milestone/30 first.

**Comment by marc0olo** [comment]
> I propose to try finalizing https://github.com/aeternity/aepp-sdk-js/milestone/30 first before releasing this ver.

agree. and it should include the migration guide from `8.x.x` => `9.x.x`

**Comment by marc0olo** [comment]
the breaking changes here are auto-generated, right?

@subhod-i can you summarize which changes you made are breaking existing functionality? we should add this to the list. I see for example the delegation signature thing missing here

**Comment by subhod-i** [comment]
> > the breaking changes here are auto-generated, right?
> > @subhod-i can you summarize which changes you made are breaking existing functionality? we should add this to the list. I see for example the delegation signature thing missing here
> 
> Auto-generated via standard-version yes. I've updated the PR description as well. But yeah, if @subhod-i can give it a bit of a human touch will be nice.

Sure. I will update it accordingly. Over `delegation` we have major changes.

**Comment by marc0olo** [comment]
@subhod-i let us know when you reviewed this :)

I am wondering about the description `change t`. that commit was about changing the default value of the gas limit as it seems

**Comment by subhod-i** [comment]
> @subhod-i let us know when you reviewed this :)
> 
> I am wondering about the description `change t`. that commit was about changing the default value of the gas limit as it seems

Yeah, I am a bit careful here. Gas Limit documentation and delegation docs we have updated. We will make sure everything is reflected as part of the release. It should have been picked by the CHANGELOG. Anyway, let me add these appropriately.

**Comment by marc0olo** [comment]
I changed some things because they were duplicated and already listed under bug fixes.

**Review by marc0olo — APPROVED** [review]

**Review by subhod-i — APPROVED** [review]
